tufaceous-lib/src/artifact.rs

```diff
@@ -283,5 +328,18 @@ fn read_entry<R: io::Read>(
 }
 
 static FILLER_TEXT: &[u8; 16] = b"tufaceousfaketxt";
+static OXIDE_JSON_FILE_NAME: &str = "oxide.json";
 static PHASE_1_FILE_NAME: &str = "image/rom";
 static PHASE_2_FILE_NAME: &str = "image/zfs.img";
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+struct OxideMetadata {
+    v: String,
+    t: String,
+}
+
+impl OxideMetadata {
+    fn os_metadata() -> Self {
+        Self { v: "1".to_owned(), t: "os".to_owned() }
+    }
+}
```

Comment:
It might be good to use the utils crate I factored out to get this: https://github.com/oxidecomputer/helios-omicron-brand/blob/main/utils/src/metadata.rs

There is both a builder and a parse routine, the latter of which will check that the `v` and `t` fields are understood.  You could then use the `.is_os()` helper or match on `.archive_type()`.

Reply:
This is a private repo -- discussed this with @jclulow, copied over `metadata.rs` for now with the goal of using this when it's opened up.